### PR TITLE
feat: add linter dupword

### PR DESCRIPTION
**Pull request by Abirdcfly**
[dupword](https://github.com/Abirdcfly/dupword) is a linter that checks for duplicate words in the source code (usually miswritten)

Examples in real code and related issues can be viewed in https://github.com/Abirdcfly/dupword/issues/3

https://github.com/Abirdcfly/dupword

### CODE REVIEW

**Comment by ldez** [comment]
In order for a pull request adding a linter to be reviewed, the linter and the PR must follow some requirements.

### Pull Request Description

- [ ] It must have a link to the linter repository.
- [ ] It must provide a short description about the linter.

### Linter

- [ ] It must not be a duplicate of another linter or a rule of a linter. (the team will help to verify that)
- [ ] It must have a valid license and the file must contain the required information by the license, ex: author, year, etc.
- [ ] The linter repository must have a CI and tests.
- [ ] It must use [`go/analysis`](https://golangci-lint.run/contributing/new-linters/).
- [ ] It must have a valid tag, ex: `v1.0.0`, `v0.1.0`.
- [ ] It must not contain `init()`.
- [ ] It must not contain `panic()`, `log.fatal()`, `os.exit()`, or similar.
- [ ] It must not have false positives/negatives. (the team will help to verify that)
- [ ] It must have tests inside golangci-lint.

### The Linter Tests Inside Golangci-lint

- [ ] They must have at least one std lib import.
- [ ] They must work with `T=<name of the linter test file>.go make test_linters`. (the team will help to verify that)

### `.golangci.reference.yml`

- [ ] The linter must be added to the list of available linters (alphabetical case-insensitive order).
    - `enable` and `disable` options
- [ ] If the linter has a configuration, the exhaustive configuration of the linter must be added (alphabetical case-insensitive order)
    - The values must be different from the default ones.
    - The default values must be defined in a comment.
    - The option must have a short description.

### Others Requirements

- [ ] The files (tests and linter) inside golangci-lint must have the same name as the linter.
- [ ] The `.golangci.yml` of golangci-lint itself must not be edited and the linter must not be added to this file.
- [ ] The linters must be sorted in the alphabetical order (case-insensitive) in the `Manager.GetAllSupportedLinterConfigs(...)` and `.golangci.reference.yml`.
- [ ] The load mode (`WithLoadMode(...)`):
    - if the linter doesn't use types: `goanalysis.LoadModeSyntax`
    - `goanalysis.LoadModeTypesInfo` required `WithLoadForGoAnalysis()` in the `Manager.GetAllSupportedLinterConfigs(...)`
- [ ] The version in `WithSince(...)` must be the next minor version (`v1.X.0`) of golangci-lint.

### Recommendations

- [ ] The linter should not use SSA. (currently, SSA does not support generics)
- [ ] The linter repository should have a readme and linting.
- [ ] The linter should be published as a binary. (useful to diagnose bug origins)

---

The golangci-lint team will edit this comment to check the boxes before and during the review.

**This checklist does not imply that we will accept the linter.**

File: test/testdata/dupword.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//golangcitest:args -Edupword
+package testdata
+
+import "fmt"
```
Comment (Abirdcfly) [inline test/testdata/dupword.go:4]: > They must have at least one std lib import.
Update for this.

**Comment by ldez** [comment]
Hello,

your linter seems weird: I expected an automatic detection of duplicate words and not a list of words.

**Comment by Abirdcfly** [comment]
> automatic detection of duplicate words and not a list of words.


There are two reasons: First, the repetitive 'the' 'a' 'and' is not grammatically valid, and there is no false positive problem. Second, I find that repeating these words is more common.

I prefer this tool to use the `-fix` parameter to help me fix the problem automatically without error. Instead of many lines that need to add `nolint`.

**Comment by bombsimon** [comment]
I guess this is also useful for people who for whatever reason write their comments in other languages than english.

**Comment by Abirdcfly** [comment]
> Your linter seems to not work as expected: the tests are failing (and it's not related to my changes).

Sorry for the error, Have updated the code. Locally Test passed via `make test`

**Comment by Abirdcfly** [comment]
https://github.com/golangci/golangci-lint/runs/8293831034?check_suite_focus=true#step:4:1508

**Comment by ldez** [comment]
> Windows test fail, I guess it is flake？

Yes it's flaky, it will fix by my PR #3204

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**